MobSF unable to perform static scanning of IOS IPA Swift based application.

<!-- ## IMPORTANT -->
<!-- Issues are ONLY for reporting bugs and feature requests. For support, questions, queries and discussions use our slack channel. Slack Channel: https://mobsf.slack.com/ , Get Invitation: https://goo.gl/6cYU5a
-->
## EXPLANATION OF THE ISSUE

I am using MobSF for performing static analysis of IOS and Android apps. Android APK and IOS applications written in objective C works fine. But when i upload an IOS IPA file written in Swift language, the static analysis fails with following error:

**"Don't Play Around. An Error just popped in! Inappropriate argument type.
'NoneType' object does not support item assignment"**

## STEPS TO REPRODUCE THE ISSUE

```
1: Start MobSF service as a docker container in a linux machine
2: Upload the IOS IPA file written in Swift language.
3: After upload, the above error pops up
```

## CONSOLE OUTPUT

```
[INFO] 25/Mar/2019 04:47:30 - "GET / HTTP/1.1" 200 7723
[INFO] 25/Mar/2019 04:47:42 - MIME Type: application/octet-stream FILE: dvia-v2-swift.ipa
[INFO] 25/Mar/2019 04:47:42 - Performing Static Analysis of iOS IPA
[INFO] 25/Mar/2019 04:47:42 - "POST /upload/ HTTP/1.1" 200 223
[INFO] 25/Mar/2019 04:47:43 - iOS Static Analysis Started
[INFO] 25/Mar/2019 04:47:43 - iOS Binary (IPA) Analysis Started
[INFO] 25/Mar/2019 04:47:43 - Generating Hashes
[INFO] 25/Mar/2019 04:47:43 - Extracting IPA
[INFO] 25/Mar/2019 04:47:43 - Unzipping
[ERROR] 25/Mar/2019 04:47:44 -
[2019-03-25 04:47:44]
[ERROR] Unzipping Error (/root/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/shared_func.py, LINE 85 "zipptr.extract(filename, ext_path)"): 'ascii' codec can't encode characters in position 89-90: ordinal not in range(128)
[INFO] 25/Mar/2019 04:47:44 - Using the Default OS Unzip Utility.
[ERROR] 25/Mar/2019 04:47:44 -
[2019-03-25 04:47:44]
[ERROR] Unzipping Error (/root/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/shared_func.py, LINE 95 "['unzip', '-o', '-I utf-8', '-q', app_path, '-d', ext_path])"): [Errno 2] No such file or directory: 'unzip': 'unzip'
[INFO] 25/Mar/2019 04:47:44 - Get Files, BIN Plist -> XML, and Normalize
[INFO] 25/Mar/2019 04:47:44 - iOS Info.plist Analysis Started
[INFO] 25/Mar/2019 04:47:44 - Finding Info.plist in iOS Binary
[WARNING] 25/Mar/2019 04:47:44 - Cannot find Info.plist file. Skipping Plist Analysis.
[INFO] 25/Mar/2019 04:47:44 - Starting Binary Analysis
[WARNING] 25/Mar/2019 04:47:44 - MobSF Cannot find binary in /root/.MobSF/uploads/35469622303ba10a2195557a3ad1810a/Payload/DVIA-v2.app/
[WARNING] 25/Mar/2019 04:47:44 - Skipping Otool, Classdump and Strings
[INFO] 25/Mar/2019 04:47:44 - Connecting to DB
[INFO] 25/Mar/2019 04:47:44 - Saving to Database
[ERROR] 25/Mar/2019 04:47:44 -
[2019-03-25 04:47:44]
[ERROR] Saving to DB (/root/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/ios/db_interaction.py, LINE 139 "VERSION=info_dict['bundle_version_name'],"): 'bundle_version_name'
[ERROR] 25/Mar/2019 04:47:44 -
[2019-03-25 04:47:44]
[ERROR] Rendering to Template (/root/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/ios/db_interaction.py, LINE 30 "'version': info_dict["bundle_version_name"],"): 'bundle_version_name'
[ERROR] 25/Mar/2019 04:47:44 - [ERROR] 'NoneType' object does not support item assignment
[ERROR] 25/Mar/2019 04:47:44 - Internal Server Error: /StaticAnalyzer_iOS/
[ERROR] 25/Mar/2019 04:47:44 - "GET /StaticAnalyzer_iOS/?name=DVIA-v2-swift.ipa&type=ipa&checksum=35469622303ba10a2195557a3ad1810a HTTP/1.1" 500 4860


```

## CONTENTS OF LOG FILES

```
Paste the contents of logs/MobSF.log here
(NOT REQUIRED FOR FEATURE REQUEST/ENHANCEMENT)
```

<!--
BEFORE POSTING YOUR ISSUE/BUG
- These comments won't show up when you submit the issue.
- GitHub issues ARE NOT FOR SUPPORT, DISCUSSIONS AND QUESTIONS! 
- If you have questions, use our slack channel. Request invitation here https://goo.gl/6cYU5a.
- Installation problems with regards to requirements.txt or VirtualBox in not considered as a MobSF issue.
- Reproduce issue in the latest master and try to add as much detail as possible.
- If you're requesting a new feature/enhancement, explain why you'd like it to be added and it's importance.
- Search this repository (top of the page) for the issue and it has not been fixed or reported already.
- Once you open a bug, you should also provide additional information if requested. 
- Failure to do so in 25 days will result in closure of the bug without further communication.
-->

--- a/Dockerfile
+++ b/Dockerfile
@@ -16,9 +16,6 @@ ENV PDFGEN_PKGFILE="wkhtmltox_0.12.5-1.bionic_amd64.deb"
 ENV PDFGEN_URL="https://github.com/wkhtmltopdf/wkhtmltopdf/releases/download/0.12.5/${PDFGEN_PKGFILE}"
 ENV YARA_URL="https://github.com/rednaga/yara-python-1"
 
-#Postgres support is set to false by default
-ARG POSTGRES=False
-
 #Update the repository sources list
 #Install Required Libs
 #see https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run
@@ -49,7 +46,7 @@ RUN \
     python3-setuptools && \
     python3 /usr/lib/python3/dist-packages/easy_install.py pip
 
-#Install sqlite3 client and pdf generator needed dependencies
+#Install git, sqlite3 client and pdf generator needed dependencies
 RUN \
     apt install -y \
     sqlite3 \
@@ -57,11 +54,7 @@ RUN \
     libjpeg-turbo8 \
     fontconfig \
     xorg \
-    xfonts-75dpi
-
-#Install git
-RUN \
-    apt install -y \
+    xfonts-75dpi \
     git
 
 #Install wkhtmltopdf for PDF Reports
@@ -71,34 +64,32 @@ RUN wget ${PDFGEN_URL} && \
    
 #Add MobSF master
 COPY . /root/Mobile-Security-Framework-MobSF
+WORKDIR /root/Mobile-Security-Framework-MobSF
 
 #Enable Use Home Directory
-WORKDIR /root/Mobile-Security-Framework-MobSF/MobSF
-RUN sed -i 's/USE_HOME = False/USE_HOME = True/g' settings.py
+RUN sed -i 's/USE_HOME = False/USE_HOME = True/g' MobSF/settings.py
 
 #Kali fix to support 32 bit execution
-WORKDIR /root/Mobile-Security-Framework-MobSF/scripts
-RUN ./kali_fix.sh
+RUN ./scripts/kali_fix.sh
 
 #Install Dependencies
-WORKDIR /root/Mobile-Security-Framework-MobSF
 RUN pip3 install -r requirements.txt
 
-#check if Postgres support must be enabled 
-WORKDIR /root/Mobile-Security-Framework-MobSF/scripts
-RUN chmod +x ./postgres_support.sh; sync; ./postgres_support.sh $POSTGRES
+#Postgres support is set to false by default
+ARG POSTGRES=False
+#check if Postgres support needs to be enabled 
+RUN cd scripts && chmod +x postgres_support.sh; sync; ./postgres_support.sh $POSTGRES
 
 #Install apkid dependencies, and enable it 
-WORKDIR /tmp
 RUN git clone --recursive ${YARA_URL} yara-python && \
     cd yara-python && \
     python3 setup.py build --enable-dex install && \
     pip3 install apkid && \
-    rm -fr /tmp/yara-python && \
+    cd .. && \
+    rm -fr yara-python && \
     sed -i 's/APKID_ENABLED.*/APKID_ENABLED = True/' /root/Mobile-Security-Framework-MobSF/MobSF/settings.py
 
 #update apkid rules
-WORKDIR /tmp
 RUN git clone https://github.com/rednaga/APKiD.git && \
     cd APKiD && \
     python3 prep-release.py && \
@@ -121,7 +112,6 @@ RUN rm -rf /var/lib/apt/lists/* /tmp/* > /dev/null 2>&1
 #Expose MobSF Port
 EXPOSE 8000
 
-WORKDIR /root/Mobile-Security-Framework-MobSF
 RUN python3 manage.py makemigrations
 RUN python3 manage.py migrate
 

--- a/Dockerfile
+++ b/Dockerfile
@@ -27,7 +27,12 @@ RUN apt update -y && apt install -y \
     libssl-dev \
     libffi-dev \
     libxml2-dev \
-    libxslt1-dev
+    libxslt1-dev \
+    locales
+
+#set locales
+RUN locale-gen en_US.UTF-8
+ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en' LC_ALL='en_US.UTF-8'
 
 #Install Oracle JDK11 LTS
 RUN apt install -y software-properties-common && \

